test/mongoose_graphql_SUITE_data/permissions_schema.gql

```diff
@@ -8,14 +8,24 @@ directive @protected (type: ProtectionType = DEFAULT, args: [String!] = [])
 
 enum ProtectionType{
   DOMAIN
+  GLOBAL
   DEFAULT
 }
 
+scalar JID
+
 type UserQuery{
   field: String
   protectedField: String @protected
   domainProtectedField(argA: String, argB: String):
     String @protected(type: DOMAIN, args: ["argA"])
+  domainInputProtectedField(argA: String, argB: DomainInput):
+    String @protected(type: DOMAIN, args: ["argA", "argB.domain"])
+  domainListInputProtectedField(domains: [DomainInput!]):
+    String @protected(type: DOMAIN, args: ["domains.domain", "domains.domains.domain"])
```

Comment:
I think that the test is only checking `domains.domain`, so maybe we could simplify this and remove `domains.domains.domain`? It's a bit confusing.